### PR TITLE
DisputableVoting: Checkpoint settings

### PR DESCRIPTION
**Pull request by facuspagnuolo**
It is recommended to review per commit

### CODE REVIEW

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 1]
```diff
@@ -68,58 +69,57 @@ contract DisputableVoting is IForwarder, DisputableAragonApp {
...
-
-    uint64 public overruleWindow;
-    mapping (address => address) internal representatives;
+    uint256 public votesLength;                             // Number of existing votes created
+    mapping (uint256 => Vote) internal votes;               // List of votes indexed by ID
+    mapping (address => address) internal representatives;  // List of representatives indexed by voter address
 
     event StartVote(uint256 indexed voteId, address indexed creator, bytes context);
-    event CastVote(uint256 indexed voteId, address indexed voter, bool supports, uint256 stake);
+    event VoteCast(uint256 indexed voteId, address indexed voter, bool supports, uint256 stake);
```
Comment (bingen) [inline apps/voting-disputable/contracts/DisputableVoting.sol:114]: Why this change? It seems to be the opposite order of the other events (verb + direct object), like `StartVote, `ChangeSupportRequired`, etc.

File: apps/voting-disputable/contracts/DisputableVoting.sol [hunk 2]
```diff
@@ -398,111 +388,123 @@ contract DisputableVoting is IForwarder, DisputableAragonApp {
...
+    * @return caster Address of the caster of the voter's vote
     */
-    function getVoteDisputableInfo(uint256 _voteId)
-        external
-        view
-        voteExists(_voteId)
-        returns (uint256 actionId, uint64 pausedAt, uint64 pauseDuration, VoteStatus status)
-    {
+    function getCastVote(uint256 _voteId, address _voter) external view voteExists(_voteId) returns (VoterState state, address caster) {
         Vote storage vote_ = votes[_voteId];
```
Comment (bingen) [inline apps/voting-disputable/contracts/DisputableVoting.sol]: Should we check that the vote exists first?

File: apps/voting-disputable/truffle.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const TruffleConfig = require('@aragon/truffle-config-v5/truffle-config')
 
 TruffleConfig.compilers.solc.version = '0.4.24'
-TruffleConfig.compilers.solc.settings.optimizer.runs = 100
+TruffleConfig.compilers.solc.settings.optimizer.runs = 1000
```
Comment (bingen) [inline apps/voting-disputable/truffle.js:4]: :heart:

File: apps/voting-disputable/contracts/DisputableVoting.sol (hunk 1 above)
Comment (facuspagnuolo) [reply apps/voting-disputable/contracts/DisputableVoting.sol:114]: The problem was the struct, I can probably swap the struct then
